### DISCUSSION TITLE
`testifylint` run on top 1w Go repos

### DISCUSSION THREAD

**Discussion by ccoVeille** [Antonboom/testifylint#159] [Ideas]
A few weeks ago, I noticed a new project 

https://github.com/alingse/go-linter-runner/

I found it interesting, I contacted @alingse

He created an issue to launch testifylint on biggest Go projects
https://github.com/alingse/go-linter-runner/issues/2

Apparently, he started working on it.

The result are impressive, there is so much error in public code base.

**Comment by Antonboom** [comment]
Hi

Good news 👍 

1. When projects number too large its difficult to scroll this issue. Can we add linter output in expandable section?
2. Can tool replace warning with link to repo and commit to more faster search of examples?

### AS IS

Run testifylint on Repo: https://github.com/eksctl-io/eksctl got output

```
https://github.com/eksctl-io/eksctl/blob/main/pkg/cfn/builder/managed_nodegroup_test.go#L126:4: error-nil: use require.NoError
```

### TO BE

Run testifylint on Repo: https://github.com/eksctl-io/eksctl

<details>

<summary>Output...</summary>

- [pkg/cfn/builder/managed_nodegroup_test.go#L126:4](https://github.com/eksctl-io/eksctl/blob/3cbd926a42beddb0398a02ae36e2202674188a9e/pkg/cfn/builder/managed_nodegroup_test.go#L126:4): error-nil: use require.NoError

</details>

**Reply by alingse** [reply]
Yes, we can improve the tool, I will do it later.

**Reply by Antonboom** [reply]
when tool will be stabilised and todos closed I do not except opportunity to add it in testifylint's README 👌

**Comment by Antonboom** [comment]
Also, how to change linter configuration?

I propose the full featured runs:
```
--enable-all --formatter.require-f-funcs
```

**Comment by alingse** [comment]
I have publish the repo as  a github action https://github.com/marketplace/actions/go-linter-runner 
you can try to integrate it into your repo's workflow

the linter config yaml can be

```yaml
workdir: '.'
install_command: go install github.com/Antonboom/testifylint@latest
linter_command: testifylint --enable-all --formatter.require-f-funcs
includes: []
excludes: []
issue_id: 0
```

the issue comment now is the expect style

-------------------------------------

Run `makezero` on Repo: https://github.com/kedacore/keda

Got total 2 line output in action: https://github.com/alingse/go-linter-runner/actions/runs/9732143542
<details>
<summary>Expand</summary>

1. [controllers/keda/scaledjob_controller.go#L357:18](https://github.com/kedacore/keda/blob/main/controllers/keda/scaledjob_controller.go#L357:18) append to slice `triggerTypes` with non-zero initialized length at
2. [controllers/keda/scaledobject_controller.go#L585:18](https://github.com/kedacore/keda/blob/main/controllers/keda/scaledobject_controller.go#L585:18) append to slice `triggerTypes` with non-zero initialized length at

</details>

Report issue: https://github.com/kedacore/keda/issues

-------------------------------------

Any bug report or advice is welcome.
